### PR TITLE
One-page form revision

### PR DESCRIPTION
**Pull request by DominikaK**
https://jira.ez.no/browse/EC-219

"Implement custom one-page form" hidden until it can be verified and rewritten to current codebase.

### CODE REVIEW

File: docs/guide/checkout/checkout_features/rma_process.md [hunk 1]
```diff
@@ -23,9 +23,9 @@ parameters:
 ## Online cancellation form
 
 There is an online cancellation form.
-After the user submits the form, an email is sent to the admin using [SendCancellationEmailDataProcessor](../../one_page_forms/one_page_forms_api/dataprocessors/dataprocessors.md#sendcancellationemaildataprocessor).
+After the user submits the form, an email is sent to the admin using [SendCancellationEmailDataProcessor](../../forms/form_api/dataprocessors.md#sendcancellationemaildataprocessor).
```
Comment (dabrt) [inline docs/guide/checkout/checkout_features/rma_process.md]:
```suggestion
After a user submits the form, an email is sent to the admin by using [SendCancellationEmailDataProcessor](../../forms/form_api/dataprocessors.md#sendcancellationemaildataprocessor).
```

File: docs/guide/checkout/checkout_features/rma_process.md [hunk 2]
```diff
@@ -46,12 +46,12 @@ parameters:
 
 ## Online RMA form
 
-eZ Commerce also offers an online RMA form. After user submits this form, an email is sent to admin using [SendRmaEmailDataProcessor](../../one_page_forms/one_page_forms_api/dataprocessors/dataprocessors.md#sendrmaemaildataprocessor).
+eZ Commerce also offers an online RMA form. After user submits this form, an email is sent to admin using [SendRmaEmailDataProcessor](../../forms/form_api/dataprocessors.md#sendrmaemaildataprocessor).
```
Comment (dabrt) [inline docs/guide/checkout/checkout_features/rma_process.md]:
```suggestion
eZ Commerce also offers an online RMA form. 
After a user submits the form, an email is sent to the admin by using [SendRmaEmailDataProcessor](../../forms/form_api/dataprocessors.md#sendrmaemaildataprocessor).
```

File: docs/guide/forms/form_api/form_api.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
+# Form API
+
+## PreDataProcessor and DataProcessors
+
+The main services for form services are `preDataProcessor` and `dataProcessors`.
+Both implement the `AbstractDataProcessor`, so the services must implement the `execute()` method.
+
+Because every form entity is a custom class, when the form is passed to `AbstractDataProcessor`,
```
Comment (dabrt) [inline docs/guide/forms/form_api/form_api.md]:
```suggestion
Every form entity is a custom class, therefore, when the form is passed to `AbstractDataProcessor`,
```

File: docs/guide/forms/form_api/form_api.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+``` yaml
+ses_phone_validator:
+    min: 9
+    max: 15
+```
+
+### VAT number validator
+
+`VatNumberValidator` checks if the VAT number for commercial customers inside the European Union is valid.
+The [VIES web-service (SOAP)](http://ec.europa.eu/taxation_customs/vies/checkVatService.wsdl) of the European Commission is used. The condition for validation is that the VAT number contains the country code.
```
Comment (dabrt) [inline docs/guide/forms/form_api/form_api.md]:
```suggestion
The [VIES web-service (SOAP)](http://ec.europa.eu/taxation_customs/vies/checkVatService.wsdl) of the European Commission is used. 
A condition for validation is that the VAT number contains a country code.
```

File: docs/guide/forms/form_api/predataprocessors.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+|---|---|---|
+|`PreFillBuyerDataProcessor`|Buyer form|`ses.customer_profile_data.data_processor.pre_fill_buyer`|
+|`PreFillContactDataProcessor`|Contact form|`ses.customer_profile_data.data_processor.pre_fill_contact`|
+|`PreFillMyAccountDataProcessor`|`MyAccount` form|`ses.customer_profile_data.data_processor.pre_fill_my_account`|
+|`PreFillDeliveryAddressDataProcessor`|Delivery address|`siso_core.data_processor.pre_fill_delivery_address`|
+|`PreFillInvoiceDataProcessor`|Invoice address|`ses.customer_profile_data.data_processor.pre_fill_invoice`|
+
+### Configuration
+
+The `use_single_name_field` setting defines if first name and surname from the form are stored in one or two separate fields:
```
Comment (dabrt) [inline docs/guide/forms/form_api/predataprocessors.md]:
```suggestion
The `use_single_name_field` setting defines whether the first name and surname from the form are stored in one or two separate fields:
```

File: docs/guide/forms/forms.md [hunk 6]
```diff
@@ -8,21 +8,14 @@ One-page forms have several common criteria:
...
-eZ Commerce uses the [Symfony forms](http://symfony.com/doc/3.4/forms.html) as a part of the solution,
-so you can take advantage of what Symfony forms offer:
+eZ Commerce uses [Symfony forms](http://symfony.com/doc/3.4/forms.html) as a part of the solution.
 
-- easy handling with form entities
-- simple and dynamic rendering with form types
-- automatic form validation
-- validation groups and much more
+You can pre-fill the form with default values when it is loaded for the first time.
+The process that pre-fills the form is called [`preDataProcessor`](form_api/predataprocessors.md).
```
Comment (dabrt) [inline docs/guide/forms/forms.md]: I'd consider using the name "pre-data processor" nad not the actual name of the processor class here.

File: docs/guide/forms/forms.md [hunk 7]
```diff
@@ -8,21 +8,14 @@ One-page forms have several common criteria:
...
 
-Additionally, you can pre-fill the form when it is loaded for the first time.
-This gives you the opportunity to improve the user experience by offering default values.
-The process that pre-fills the form is called `preDataProcessor`.
-
-You can define processes that are executed after the form is submitted.
-These processes are services that are executed is a sequence and are able to exchange the data.
-These processes are called `dataProcessors`.
+You can define actions that are executed after the form is submitted.
+These actions are services that are called is a sequence and are able to exchange the data.
```
Comment (dabrt) [inline docs/guide/forms/forms.md]:
```suggestion
These actions are services that are called in a sequence and are able to exchange the data.
```

"exchange" -> "replace" ?

File: docs/guide/forms/forms.md [hunk 8]
```diff
@@ -120,10 +107,10 @@ parameters:
             - ses_forms.send_confirmation_data_processor
 ```
 
-|Configuration key (* required)|Description|
+|Configuration key|Description|
 |--- |--- |
-|`modelClass`|Required. Fullly-qualified class path to the form entity. This class needs to extend `Silversolutions\Bundle\EshopBundle\Entities\Forms\AbstractFormEntity`|
-|`typeClass`|Required if `typeService` is not defined. Fullly-qualified class path to the form type. Using `typeService` gives more flexibility.|
+|`modelClass`|Required. Fully-qualified class path to the form entity. This class must extend `Silversolutions\Bundle\EshopBundle\Entities\Forms\AbstractFormEntity`|
```
Comment (dabrt) [inline docs/guide/forms/forms.md]:
```suggestion
|`modelClass`|Required. A fully-qualified class path to the form entity. This class must extend `Silversolutions\Bundle\EshopBundle\Entities\Forms\AbstractFormEntity`|
```

File: docs/guide/forms/forms.md [hunk 9]
```diff
@@ -120,10 +107,10 @@ parameters:
...
+|`typeClass`|Required if `typeService` is not defined. Fully-qualified class path to the form type. Using `typeService` gives more flexibility.|
```
Comment (dabrt) [inline docs/guide/forms/forms.md]:
```suggestion
|`typeClass`|Required if `typeService` is not defined. A fully-qualified class path to the form type. Using `typeService` gives more flexibility.|
```

**Review by dabrt — CHANGES_REQUESTED** [review]

File: docs/guide/forms/forms.md (hunk 6 above)
Comment (DominikaK) [inline docs/guide/forms/forms.md]:
```suggestion
The process that pre-fills the form is called a [pre-data processor](form_api/predataprocessors.md).
```

**Review by dabrt — APPROVED** [review]

**Pull request merged**